Lib/compositor/logger.py

```diff
@@ -16,7 +16,10 @@
 The returned log is in XML format.
 """
 
-from cStringIO import StringIO
+try:
+    from io import StringIO
```

Comment:
The `io` module is present on both Python 2 and 3, so this does never raises `ImportError`. I think you wanted to do the opposite: i.e. attempt to import cStringIO (python2 only), and if that fails, use io.StringIO (for python 3)?

Note that `cStringIO.StringIO` can take either bytes or unicode (the latter are encoded using the default encoding), whereas the `io.StringIO` only accepts unicode strings.